files/external_node_v2.rb

```diff
@@ -206,8 +210,15 @@ def upload_facts(certname, req)
   begin
     res = initialize_http(uri)
     res.read_timeout = SETTINGS[:timeout]
-    res.start { |http| http.request(req) }
-    cache("#{certname}-push-facts", "Facts from this host were last pushed to #{uri} at #{Time.now}\n")
+    res.start do |http|
+      response = http.request(req)
+      if response.code.start_with?('2')
+        cache("#{certname}-push-facts", "Facts from this host were last pushed to #{uri} at #{Time.now}\n")
+      else
+        puts "Server responded with: #{response.code} #{response.message}"
```

Comment:
Sending output to stdout will probably interfere with the ENC output, should this be to stderr or a log file?